src/pybind/mgr/mgr_util.py

```diff
@@ -627,70 +626,59 @@ def create_self_signed_cert(organisation: str = 'Ceph',
-    for k, v in dname.items():
-        setattr(subj, k, v)
+    import json
+    import subprocess
+
+    private_key = subprocess.run(["python3", "-m", "ceph.pybind.mgr.cryptotools", "create_self_signed_cert", "--private_key"],
+                                 capture_output=True)
+
+    pkey = private_key.stdout.strip().decode('utf-8')
+
+    data = {"dname": dname, "private_key": pkey}
 
-    # create a self-signed cert
-    cert = crypto.X509()
-    cert.set_subject(req.get_subject())
-    cert.set_serial_number(int(uuid4()))
-    cert.gmtime_adj_notBefore(0)
-    cert.gmtime_adj_notAfter(10 * 365 * 24 * 60 * 60)  # 10 years
-    cert.set_issuer(cert.get_subject())
-    cert.set_pubkey(pkey)
-    cert.sign(pkey, 'sha512')
+    result = subprocess.run(["python3", "-m", "ceph.pybind.mgr.cryptotools", "create_self_signed_cert", "--certificate"],
+                            input=json.dumps(data).encode("utf-8"),
+                            capture_output=True)
 
-    cert = crypto.dump_certificate(crypto.FILETYPE_PEM, cert)
-    pkey = crypto.dump_privatekey(crypto.FILETYPE_PEM, pkey)
+    # Check result with a CompletedProcess
+    if result.returncode != 0 or result.stderr != b'':
+        raise ValueError(result.stderr)
 
-    return cert.decode('utf-8'), pkey.decode('utf-8')
+    cert = result.stdout.strip().decode('utf-8')
+    return cert, pkey
 
 
 def verify_cacrt_content(crt):
     # type: (str) -> int
-    from OpenSSL import crypto
+
     try:
-        crt_buffer = crt.encode("ascii") if isinstance(crt, str) else crt
-        x509 = crypto.load_certificate(crypto.FILETYPE_PEM, crt_buffer)
-        no_after = x509.get_notAfter()
-        if not no_after:
-            raise ServerConfigException("Certificate does not have an expiration date.")
-
-        end_date = datetime.datetime.strptime(no_after.decode('ascii'), '%Y%m%d%H%M%SZ')
-        if x509.has_expired():
-            org, cn = get_cert_issuer_info(crt)
-            msg = f'Certificate issued by "{org}/{cn}" expired on {end_date}'
-            logger.warning(msg)
-            raise ServerConfigException(msg)
-
-        # Certificate still valid, calculate and return days until expiration
-        return (end_date - datetime.datetime.utcnow()).days
-
-    except (ValueError, crypto.Error) as e:
+        import subprocess
+        result = subprocess.run(["python3", "-m", "ceph.pybind.mgr.cryptotools", "verify_cacrt_content"],
+                                input=crt if isinstance(crt, bytes) else crt.encode('utf-8'),
+                                capture_output=True)
+        # The above script will only produce stdout output.
+        # The only scenarios that produce stderr output are failures to import modules
+        # or syntax errors which test_tls.py will catch
+
+        # Check result of CompletedProcess
+        if result.returncode != 0 or result.stderr != b'':
+            logger.warning(result.stderr)
+            raise ValueError(result.stderr)
```

Comment:
Given this structure (`run ; if .returncode ...`) repeats a lot, I'd wrap it inside a helper method.